cmd/cloud_sql_proxy/proxy.go

```diff
@@ -328,16 +328,21 @@ func parseInstanceConfigs(dir string, instances []string, cl *http.Client) ([]in
 
 // CreateInstanceConfigs verifies that the parameters passed to it are valid
 // for the proxy for the platform and system and then returns a slice of valid
-// instanceConfig.
+// instanceConfig. It is possible for the instanceConfig to be empty if no valid
+// configurations were specified, however `err` will be set.
 func CreateInstanceConfigs(dir string, useFuse bool, instances []string, instancesSrc string, cl *http.Client) ([]instanceConfig, error) {
+	// Used to signify that there was no valid parsed config
+	var emptyConfigs []instanceConfig
+
 	if useFuse && !fuse.Supported() {
-		return nil, errors.New("FUSE not supported on this system")
+		return emptyConfigs, errors.New("FUSE not supported on this system")
 	}
 
 	cfgs, err := parseInstanceConfigs(dir, instances, cl)
 	if err != nil {
 		// Error when unable to correctly parse the instance configuration
-		return nil, err
+		// However, return the configuration of what was parsed correctly
+		return cfgs, err
```

Comment:
This is the core of the change -- we're returning what we could parse right here and then using that above.

Reply:
I don't think this is a correct way to fix this. 

If we return here and `cfgs` has a valid value, then the rest of this function is ignored and the other flags that might have been set are ignored and not processed correctly( like `-dir` or `-fuse`).

Reply:
Thanks Kurtis. I completely forgot about all those other potential things.

Would it make more sense to aggregate the configurations and errors?

Reply:
I'm not sure.

Thinking about this more in-depth with the current code path, a `-skip-startup-validation` flag might not be the behavior we want to satisfy the problem. Originally when @broady and I had discussed this issue we had discussed just separating the parsing of the config with the validation that it was valid. However I think to satisfy the App Engine team we actually just want to parse the instances individually and continue running as long as one instance was parsed successfully.

So perhaps a better solution would be to let the flag be something like `-skip-failed-instance-parses`, and then check the flag inside `parseInstanceConfigs` level, after each individual `parseInstanceConfig`, and only return values for instances that are parsed successfully.

Reply:
That would be the MVP I guess for our flex team unblocked. I think I can do that and move some things around. Let me try it out and see how it looks.

Reply:
When you say parse properly, do you mean something that does not return an error from `parseInstanceConfig`? I don't think checking the connection string would be enough -- since some customers might have "dead connection strings" that point to deleted instances.

Reply:
Yes, something that does not return an error from parseInstanceConfig. Maybe `-skip-failed-instance-config` would be better?

Reply:
Yeah, that sounds good. I just did the refactoring; I agree with you looking at this more now.

There's no need to burden the CLI utility with needing to know how to handle errors and conditionals. Makes for a messy interface.

I have something working; just refining tests and doing some manual testing.